data/json/vehicleparts/vp_flags.json

```diff
@@ -3,7 +3,9 @@
     "id": "ALTERNATOR",
     "type": "json_flag",
     "context": [ "vehicle_part" ],
-    "info": "An alternator.  When mounted on a gasoline or diesel engine and the engine is on, will produce electrical power that can be stored in a battery."
+    "info": "An alternator.  When mounted on a gasoline or diesel engine and the engine is on, will produce electrical power that can be stored in a battery.",
+    "requires_flag": "ENGINE",
+    "cant_remove": "Remove attached alternator first."
```

Comment:
You also need to update `lang/extract_json_strings.py` to extract this string for translation.

Reply:
Do I just add

 ```
    if "cant_remove" in item:
       writestr(outfile, item["cant_remove"], **kwargs)
       wrote = True
```

into `def extract(item, infilename):`?

Reply:
I think so. You can run `lang/update_pot.sh` to see if it correctly extracts these strings.

Reply:
It should add them to the po files in `\lang\po`?
Either it is not working or I don't know how it should be used. Maybe both.

Reply:
`lang/update_pot.sh` updates lang/po/cataclysm-dda.pot.
You can run `lang/merge_po.sh` after that to update `lang/po/*.po`.
See https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/TRANSLATING.md#maintainers